### PR TITLE
fix(commit): Adds support for multilines on Windows

### LINKED ISSUE #286: Messages with newlines are broken on Windows
When commiting, when messages are split into lines, `git` will consider each line a new paragraph, meaning that paragraphs that are broken into many lines, such as [cz-conventional-changelog](https://github.com/commitizen/cz-conventional-changelog) emits, will result in an unexpected formatting of the final message.

I opened stevelacy/gulp-git#122 which allows gulp-git to pass multiline messages as expected, but to prevent breaking changes, it requires the message as either an array of paragraphs and the `multiline` option set to `true` or a string with `\n`s. It seems that simply changing [this line](https://github.com/commitizen/cz-cli/blob/master/src/git/commit.js#L21):
```javascript
operatingSystemNormalizedMessage = escapedMessage.split(/\r?\n/);
```
with this:
```javascript
operatingSystemNormalizedMessage = escapedMessage.replace('\r', '');
```
would solve the issue, unless I'm missing something.

I'll try to open a PR later when I have a chance, unless someone else wants to do it.

Thanks.

### PR DESCRIPTION
**Pull request by rtomchinsky**
Closes #286

### CODE REVIEW

**Comment by LinusU** [comment]
Would this be fixed by #256?

**Comment by rtomchinsky** [comment]
For what I can see, I think it would actually also break on other OSes, since that just makes git interpret all `\n`s as a new paragraph. If you pass the message into the stdin of the child process, with `git commit -F -` (see stevelacy/gulp-git@72cb808521231a970cdd15ac86aa07cef6265435), it would work as expected, while still removing gulp-git as a dependency.

File: src/git/commit.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ function commit(sh, repoPath, message, options, done) {
   // On windows we must use an array in gulp-git instead of a string because
   // command line parsing works differently
   if(os.platform()=="win32") {
-    operatingSystemNormalizedMessage = escapedMessage.split(/\r?\n/);
+    operatingSystemNormalizedMessage = escapedMessage.replace(/\r/g, '');
```
Comment (pmcelhaney) [inline src/git/commit.js]: Did something in gulp-git change so that it now works with a string? Or is the above comment incorrect?

Comment (rtomchinsky) [inline src/git/commit.js]: gulp-git 1.9.0 added support for multi-line messages. I guess I forgot to change that comment

**Comment by jimthedev** [comment]
I am going to close this for now since we don't use gulp-git anymore.

**Pull request closed without merging**